components/br-stream/src/endpoint.rs

```diff
@@ -143,26 +167,49 @@ where
                 .with_label_values(&["get_router_lock"])
                 .observe(sw.lap().as_secs_f64());
             let mut kv_count = 0;
+            let total_size = kvs.as_slice().iter().fold(0usize, |init, kv| init + kv.size());
+            metrics::HEAP_MEMORY
+                .with_label_values(&["alloc"])
+                .inc_by(total_size as f64);
             for kv in kvs {
+                let size = kv.size();
                 // TODO build a error handle mechanism #error 6
                 if kv.should_record() {
                     if let Err(err) = router.on_event(kv).await {
-                        error!("backup stream failed in backup batch"; "error" => ?err);
+                        err.report(format!("failed to send event."));
                     }
+                    metrics::HEAP_MEMORY
+                        .with_label_values(&["free"])
+                        .inc_by(size as f64);
                     kv_count += 1;
                 }
             }
             HANDLE_KV_HISTOGRAM.observe(kv_count as _);
+            let time_cost = sw.lap().as_secs_f64();
+            if time_cost > 120.0 {
```

Comment:
👀

Reply:
If there is a hot write region, the lock may be hold too long. (Up to 1000s if the hot region keep writing....)

Reply:
@YuJuncen i mean 120 is a magic constant